pvlib/spectrum/mismatch.py

```diff
@@ -461,35 +479,25 @@ def spectral_factor_firstsolar(precipitable_water, airmass_absolute,
        MMF Approach, TUV Rheinland Energy GmbH report 21237296.003,
        January 2017
     """
-
-    # --- Screen Input Data ---
-
-    # *** Pw ***
-    # Replace Pw Values below 0.1 cm with 0.1 cm to prevent model from
-    # diverging"
     pw = np.atleast_1d(precipitable_water)
     pw = pw.astype('float64')
     if np.min(pw) < min_precipitable_water:
         pw = np.maximum(pw, min_precipitable_water)
-        warn('Exceptionally low pw values replaced with '
-             f'{min_precipitable_water} cm to prevent model divergence')
+        warn('Low precipitable water values replaced with '
+             f'{min_precipitable_water} cm in the calculation of spectral '
+             'mismatch.')
 
-    # Warn user about Pw data that is exceptionally high
     if np.max(pw) > max_precipitable_water:
         pw[pw > max_precipitable_water] = np.nan
-        warn('Exceptionally high pw values replaced by np.nan: '
-             'check input data.')
-
-    # *** AMa ***
-    # Replace Extremely High AM with AM 10 to prevent model divergence
-    # AM > 10 will only occur very close to sunset
-    if np.max(airmass_absolute) > 10:
-        airmass_absolute = np.minimum(airmass_absolute, 10)
-
-    # Warn user about AMa data that is exceptionally low
-    if np.min(airmass_absolute) < 0.58:
-        warn('Exceptionally low air mass: ' +
-             'model not intended for extra-terrestrial use')
+        warn('High precipitable water values replaced with np.nan in '
+             'the calculation of spectral mismatch.')
+
+    airmass_absolute = np.minimum(airmass_absolute, max_airmass_absolute)
+
+    if np.min(airmass_absolute) < min_airmass_absolute:
+        airmass_absolute = np.maximum(airmass_absolute, min_airmass_absolute)
+        warn('Low airmass values replaced with 'f'{min_airmass_absolute} in '
+             'the calculation of spectral mismatch.')
         # pvl_absoluteairmass(1,pvl_alt2pres(4340)) = 0.58 Elevation of
```

Comment:
May as well update this to pvlib-python functions of just remove it?

Reply:
@echedey-ls I don't follow, what do you mean?

Reply:
Regarding the comment on line 501 in your branch that says `pvl_absoluteairmass(1,pvl_alt2pres(4340)) = 0.58`

And your previous comment about removing references to pvlib matlab: https://github.com/pvlib/pvlib-python/pull/2100#issuecomment-2205942534

I think we can update this comment to use pvlib python functions since it is somehow misleading to someone that does not know about pvlib matlab. Updating or just removing, thou I prefer updating.

```suggestion
        # pvlib.atmosphere.get_absolute_airmass(1,pvlib.atmosphere.alt2pres(4340)) = 0.58 Elevation of
```

(well, without `pvlib.atmosphere.` is probably good, and IDK if flake8 will complain)

Reply:
Ahh okay, understood. Good point. I have made the change with a line break in the last commit. Thanks